### PR TITLE
Fil 753 allow bootstrap to apply subsets of migrations

### PR DESCRIPTION
**Pull request by geoidesic**

### CODE REVIEW

**Comment by geoidesic** [comment]
Please review the docs also

File: README.rst [hunk 1]
```diff
@@ -80,3 +83,54 @@ a-json-file.json
...
 
+
+==============
+Data structure
+==============
+Note that it is possible to conform to DRY principles *to a degree*. If you wish to accumulate data, then records in later files will still need to be defined but they do not require all the data to be present, providing that your component uses `get_or_create` when generating new records.
+
+E.g. Let's say you have two json files to import: `roles.json` and `demo.json`. You may define all the permissions and roles in the first file. Any records which may be required in the second just need their unique identifier property, so that `get_or_create` will execute a SELECT instead of an INSERT.
+
+Note that each root key in your json data file must correspond by name with one of the components.
```
Comment (chaosk) [inline README.rst]: Actually, it needs to correspond to Component.field_name attribute set in your component class. Also, you can technically read from other keys as well by using `self.bootstrap.data("other_key")`

File: README.rst [hunk 2]
```diff
@@ -80,3 +83,54 @@ a-json-file.json
...
+And the json data type (i.e. list or object) for a given root key should correspond to the `default_factory` 
+
+Thus in `roles.json` you might have this structure:
+
+.. code-block:: json
```
Comment (chaosk) [inline README.rst]: This is Sphinx-specific syntax, maybe move most of this file to docs/ folder that you can base off other projects?

File: tests/test_components.py [hunk 3]
```diff
@@ -22,6 +22,7 @@
 from djangocms_moderation.models import ModerationCollection, Role, Workflow
 from djangocms_versioning.constants import DRAFT, PUBLISHED
 
+import sys
```
Comment (chaosk) [inline tests/test_components.py]: Unnecessary import

File: README.rst (hunk 1 above)
Comment (geoidesic) [reply README.rst]: Isn't it `self.bootstrap.data["other_key"]`? (square brackets)... at least that's what we decided was best in collections.py

File: README.rst (hunk 2 above)
Comment (geoidesic) [reply README.rst]: I don't know how that works and I want to get this merged. We can move docs once everyone understands how that works.

**Pull request merged**